globus_cli/commands/endpoint/server/delete.py

```diff
@@ -69,9 +69,22 @@ def server_delete(endpoint_id, server):
 
     mode = _detect_mode(server)
 
-    if mode != 'id':
-        endpoint, server_list = get_endpoint_w_server_list(endpoint_id)
+    # list (even if not necessary) in order to make errors more consistent when
+    # mode='id'
+    endpoint, server_list = get_endpoint_w_server_list(endpoint_id)
+
+    if server_list == 'S3':
+        raise click.UsageError('You cannot delete servers from S3 endpoints.')
 
+    # GCPs will have one server in their list, with a null hostname
+    # we don't *have to* raise an error in this case, since the API would deny
+    # it too, but doing so makes our errors a little bit more consistent
+    # with deletes against S3 endpoints and shares
+    if not all(x['hostname'] is None for x in server_list):
```

Comment:
Is this check correct?

I think you want `server_list and all(x['hostname'] is None for x in server_list)` or maybe `len(server_list) == 1 and server_list[0]['hostname'] is None` or maybe even just `endpoint['is_globus_connect']`.

Reply:
You caught me before I squashed this.
I meant `all()` as opposed to `not all()`, yes.

`endpoint['is_globus_connect']` is cleaner though. Should I change to that?